src/pages/portfolio/ui/assets-table.tsx

```diff
@@ -121,287 +94,351 @@ const NotConnectedNotice = () => {
-        'Error formatting amount for',
-        metadata.symbol,
-        'amount:',
-        amount.toJsonString(),
-        'error:',
-        error,
-      );
-      return { value: 0, balance, hasError: true };
-    }
-  });
-
-  const totalValue = valuesWithMetadata.reduce((acc, { value }) => acc + value, 0);
-
-  if (totalValue === 0) {
-    return { distribution: [], sortedBalances: [] };
-  }
-
-  // Sort by value percentage in descending order and calculate distribution
-  const sorted = valuesWithMetadata
-    .map((item, index) => {
-      const metadata = getMetadataFromBalancesResponse.optional(item.balance);
-      const primaryColor = metadata?.images[0]?.theme?.primaryColorHex;
-      return {
-        ...item,
-        percentage: (item.value / totalValue) * 100,
-        color:
-          primaryColor ??
-          `hsl(${(index * GOLDEN_RATIO_ANGLE) % 360}, ${COLOR_SATURATION}%, ${COLOR_LIGHTNESS}%)`,
-      };
-    })
-    .sort((a, b) => b.percentage - a.percentage);
-
-  // Split into main assets and small assets
-  const mainAssets = sorted.filter(asset => asset.percentage >= SMALL_ASSET_THRESHOLD);
-  const smallAssets = sorted.filter(asset => asset.percentage < SMALL_ASSET_THRESHOLD);
-
-  const otherPercentage = smallAssets.reduce((acc, asset) => acc + asset.percentage, 0);
-
-  const distribution: AssetDistribution[] = [
-    ...mainAssets.map(({ percentage, color, hasError }) => ({
-      percentage,
-      color,
-      hasError,
-      isOther: false,
-    })),
-    ...(otherPercentage > 0
-      ? [
-          {
-            percentage: otherPercentage,
-            color: `hsl(0, 0%, ${COLOR_LIGHTNESS}%)`,
-            hasError: false,
-            isOther: true,
-          },
-        ]
-      : []),
-  ];
-
-  return {
-    distribution,
-    sortedBalances: sorted.map(({ balance }) => balance),
-  };
-};
+  },
+);
 
 export const AssetsTable = observer(() => {
-  const { connected } = connectionStore;
-  const addressIndex = new AddressIndex({ account: connectionStore.subaccount });
-  const { data: balances, isLoading: balancesLoading } = useBalances(addressIndex.account);
-  const { data: assets, isLoading: assetsLoading } = useAssets();
-  const [distribution, setDistribution] = useState<DistributionResult>();
+  const { unifiedAssets, isLoading, isPenumbraConnected, isCosmosConnected } = useUnifiedAssets();
+  const [expandedRows, setExpandedRows] = useState<Record<string, boolean>>({});
 
-  useEffect(() => {
-    if (balances) {
-      setDistribution(calculateAssetDistribution(balances));
-    }
-  }, [balances]);
+  const { data: prices } = useAssetPrices(unifiedAssets.map(asset => asset.symbol));
 
-  const isLoading = balancesLoading || assetsLoading || !balances || !assets || !distribution;
+  const toggleRow = (symbol: string) => {
+    setExpandedRows(prev => ({
+      ...prev,
+      [symbol]: !prev[symbol],
+    }));
+  };
 
   if (isLoading) {
     return <LoadingState />;
   }
```

Comment:
question: is it possible to independently display shielded assets and public loading skeletons, and after load display all assets? Just curious if this UX is possible and if it could improve initial content load time?

Reply:
It is, requires some rework of the hooks, noted for the next PR.